subprojects/docs/src/docs/userguide/workingWithFiles.adoc

```diff
@@ -15,47 +15,288 @@
+    include "*.pdf"
+    into "${buildDir}/toArchive"
+}
+----
+
+One thing to note in this example is that only PDFs directly in the `reports` directory are copied. If there are PDFs in subdirectories of `reports`, they'll be left out. You can change that by using an Ant-style glob pattern:
+
+----
+task copyReports(type: Copy) {
+    from "${buildDir}/reports"
+    include "**/*.pdf"
+    into "${buildDir}/toArchive"
+}
+----
+
+This approach has the side effect of copying the directory structure below `reports` as well as the files. This may be what you want. It may not be. Copying files includes many such decisions and there are too many to cover here, but know that there are very few requirements that can't be handled elegantly by Gradle copy specifications.
+
+Head over to ~TODO~ in order to learn all the different options available for getting the exact behavior you want.
+
+=== Copying directory hierarchies
+
+You may have the need to copy not just files, but the directory structure they reside in as well. This is, in fact, the default behavior when you specify a directory as the `from()` argument. So the following will copy everything in the `reports` directory, including all its subdirectories:
+
+----
+task copyReports(type: Copy) {
+    from "${buildDir}/reports"
+    into "${buildDir}/toArchive"
+}
+----
+
+The key aspect that users struggle with is controlling how much of the directory structure goes to the destination. In the above example, do you get a `toArchive/reports` directory or does everything in `reports` go straight into `toArchive`? The answer is the latter. If a directory is part of the `from()` path, then it *won't* appear in the destination.
+
+So how do you ensure that `reports` itself is copied across, but not any other directory in `$buildDir`? The answer is to add it as an include pattern:
+
+----
+task copyReports(type: Copy) {
+    from("${buildDir}") {
+        include "reports/**"
+    }
+    into "${buildDir}/toArchive"
+}
+----
+
+You'll get the same behavior as before except with one extra level of directory in the destination, i.e. `toArchive/reports`.
+
+One thing to note is how the `include()` directive applies only to the `from()`, whereas the directive in the previous section applied to the whole task. These different levels of granularity in the copy specification allow you to easily handle most requirements that you will come across. TODO: Link to more details on copy specs.
+
+=== Creating archives (zip, tar, etc.)
+
+From the perspective of Gradle, packing files into a single archive is effectively a copy in which the destination is an archive file instead of a directory on the file system. This means that archiving files looks a lot like copying, with all of the same features!
+
+The simplest case involves packing all the contents of a directory into an archive, such as a zip in this example:
+
+----
+task packageDistribution(type: Zip) {
+    archiveName = "my-distribution.zip"
```

Comment:
Maybe we should promote the use of the `base` plugin here instead as it provides the conventions.

Reply:
Since there are two examples, I'd like to have one without the plugin, to demonstrate that it's not needed, and one with to show the convenience. I'll probably trim this section a bit too, as I've now updated the in-depth section on creating archives which covers the `base` plugin and naming conventions in detail.

Reply:
- [ ] Use one sample without the Base Plugin, and one with